src/OpenLoco/src/GameCommands/RemoveTown.cpp

```diff
@@ -0,0 +1,161 @@
+#include "Audio/Audio.h"
+#include "Economy/Expenditures.h"
+#include "GameCommands.h"
+#include "Localisation/StringIds.h"
+#include "Localisation/StringManager.h"
+#include "Map/BuildingElement.h"
+#include "Map/RoadElement.h"
+#include "Map/TileElement.h"
+#include "Map/TileLoop.hpp"
+#include "Map/TileManager.h"
+#include "MessageManager.h"
+#include "Objects/ObjectManager.h"
+#include "S5/S5.h"
+#include "Ui/WindowManager.h"
+#include "ViewportManager.h"
+#include "World/IndustryManager.h"
+#include "World/StationManager.h"
+#include "World/TownManager.h"
+
+using namespace OpenLoco::Interop;
+using namespace OpenLoco::World;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x0049711F
+    static uint32_t removeTown(const TownRemovalArgs& args, const uint8_t flags)
+    {
+        for (auto& station : StationManager::stations())
+        {
+            if (station.town == args.townId)
+            {
+                setErrorText(StringIds::all_stations_near_this_town_must_be_removed_first);
+                return FAILURE;
+            }
+        }
+
+        if (!(flags & Flags::apply))
+        {
+            return 0;
+        }
+
+        // NB: vanilla did not set an expenditute type
+        GameCommands::setExpenditureType(ExpenditureType::Construction);
+
+        // Iterate over the entire map to find town tiles
+        // TODO: can't we do better than vanilla for this? e.g. a radius around the town centre?
+        TilePosRangeView tileLoop{ { 1, 1 }, { kMapColumns - 1, kMapRows - 1 } };
+
+        for (auto& tilePos : tileLoop)
+        {
+            auto tile = TileManager::get(tilePos);
+
+            for (auto& element : tile)
+            {
+                if (element.isGhost())
+                {
+                    continue;
+                }
+
+                auto* buildingEl = element.as<BuildingElement>();
+                if (buildingEl != nullptr)
+                {
+                    if (buildingEl->has_40())
+                    {
+                        continue;
+                    }
+
+                    if (buildingEl->multiTileIndex() != 0)
+                    {
+                        continue;
+                    }
+
+                    auto worldPos = toWorldSpace(tilePos);
+                    auto maybeTown = TownManager::getClosestTownAndDensity(worldPos);
+                    if (maybeTown)
+                    {
+                        TownId nearestTown = maybeTown->first;
+                        if (nearestTown != args.townId)
+                        {
+                            continue;
+                        }
+                    }
+
+                    BuildingRemovalArgs rmArgs{};
+                    rmArgs.pos = Pos3(worldPos.x, worldPos.y, buildingEl->baseHeight());
+                    if (doCommand(rmArgs, flags) == FAILURE)
+                    {
+                        // Stop processing this tile
```

Comment:
This isn't correct vanilla didn't care about FAILURE i think you've confused it with the reset tile processing after any removal.

Reply:
So just disregard the doCommand result and continue with the next element then?

Reply:
The result is meant to be used to inform you to reset the tile element loop so
```
if (doCommand(rmArgs, flags) != FAILURE)
{ 
    resetLoop = true;
    break;
}```